lib/samly/auth_handler.ex

```diff
@@ -59,7 +59,9 @@ defmodule Samly.AuthHandler do
 
       _ ->
         relay_state = State.gen_id()
-        {idp_signin_url, req_xml_frag} = Helper.gen_idp_signin_req(sp, idp_rec)
+
+        {idp_signin_url, req_xml_frag} =
```

Comment:
```suggestion
        {idp_signin_url, req_xml_frag} = Helper.gen_idp_signin_req(sp, idp_rec, nameid_format)
```

Get the `nameid_format` as part of the first destructing assignment in this function.

```
    %IdpData{id: idp_id, nameid_format: nameid_format} = idp = conn.private[:samly_idp]
```